scripts/firedrake-install

```diff
@@ -1048,15 +1036,17 @@ def build_and_install_petsc():
         petsc_dir, petsc_arch = get_petsc_dir()
         check_call(["rm", "-rf", os.path.join(petsc_dir, petsc_arch)])
         petsc_options = get_petsc_options()
+        # As of 2022-03-17, it looks like setting MACOSX_DEPLOYMENT_TARGET on macOS 12.3 system actually breaks petsc configuration
```

Comment:
This has now been fixed in a hopefully better way in #2408. I claim we don't want to delete this code.